api/src/main/java/org/geysermc/geyser/api/item/custom/v2/component/java/Consumable.java

```diff
@@ -0,0 +1,153 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.api.item.custom.v2.component.java;
+
+import org.checkerframework.checker.index.qual.Positive;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.common.returnsreceiver.qual.This;
+import org.geysermc.geyser.api.GeyserApi;
+import org.geysermc.geyser.api.util.GenericBuilder;
+
+/**
+ * The consumable component is used to mark
+ * an item as consumable. Further, it allows specifying
+ * the consume duration and animation to play when consuming.
+ */
+public interface Consumable {
+
+    /**
+     * The seconds it takes to consume the item.
+     * This it the amount of time the animation will play for. Defaults to {@code 1.6}.
+     *
+     * @return the consume duration, in seconds
+     */
+    @Positive float consumeSeconds();
+
+    /**
+     * The animation that should play when consuming the item. Defaults to {@link Animation#EAT}.
+     * 
+     * @return the animation to play
+     */
+    @NonNull Animation animation();
+
+    /**
+     * Creates a builder for the consumable component.
+     *
+     * @return a new builder
+     */
+    static Builder builder() {
+        return GeyserApi.api().provider(Consumable.Builder.class);
+    }
+
+    /**
+     * Creates a consumable component.
+     *
+     * @param consumeSeconds the consume duration, in seconds
+     * @param animation the animation to play when consuming
+     * @return the consumable component
+     */
+    static Consumable of(float consumeSeconds, Animation animation) {
+        return Consumable.builder().consumeSeconds(consumeSeconds).animation(animation).build();
+    }
+
+    /**
+     * Not all animations work perfectly on bedrock. Bedrock behaviour is noted per animation. The {@code toot_horn} animation doesn't exist on bedrock, and is therefore not listed here.
```

Comment:
Maybe worth linking to https://wiki.bedrock.dev/items/item-components#use-animations & specifying the Bedrock client version used to find these